packages/core/snapshotcreator/snapshotcreator.go

```diff
@@ -104,7 +106,7 @@ func CreateSnapshot(opts ...options.Option[Options]) error {
 	if err != nil {
 		return err
 	}
-	if _, _, err = engineInstance.NotarizationManager.Attestations.Commit(0); err != nil {
+	if _, _, err = engineInstance.Notarization.Attestations().(*slotnotarization.Attestations).Commit(0); err != nil {
```

Comment:
Why not add these to the interface? To keep it lean as it's only needed here and in some tests?

Reply:
Yes. The setters are currently only used in tests and during snapshot creation.